### PR TITLE
New Device (Matter Lock)Aqara U200

### PR DESCRIPTION
**Pull request by lelandblue**
This is a thread-based Matter lock seeking WWST certification.

### CODE REVIEW

**Comment by lelandblue** [comment]
@ctowns Will we automatically switch to a profile that supports lock codes? I believe this device supports lock codes, however that was not the profile selected during the certification process.

**Comment by ctowns** [comment]
> @ctowns Will we automatically switch to a profile that supports lock codes? I believe this device supports lock codes, however that was not the profile selected during the certification process.

We will not automatically switch to a profile that supports lock codes. We currently only support switching to a profile _without lock codes_ because the generic `matter-lock` fingerprint matches to a profile with lock codes (base-lock).

Also, if we know the device does support lock codes, can we override the profile that was selected during the WWST submission? I think it would probably be best if manufacturer fingerprints matched to the correct profile if possible without relying on profile switching in the driver if we can. What do you think?

**Comment by lelandblue** [comment]
> > @ctowns Will we automatically switch to a profile that supports lock codes? I believe this device supports lock codes, however that was not the profile selected during the certification process.
> 
> We will not automatically switch to a profile that supports lock codes. We currently only support switching to a profile _without lock codes_ because the generic `matter-lock` fingerprint matches to a profile with lock codes (base-lock).
> 
> Also, if we know the device does support lock codes, can we override the profile that was selected during the WWST submission? I think it would probably be best if manufacturer fingerprints matched to the correct profile if possible without relying on profile switching in the driver if we can. What do you think?

Ah - we are cautious about that for good reasons. Yes, I do think we should push back on the existing submission and recommend that they use a different profile.

File: drivers/SmartThings/matter-lock/fingerprints.yml [hunk 1]
```diff
@@ -1,4 +1,10 @@
 matterManufacturer:
+#Aqara
+  - id: 4447/10242
+    deviceLabel: Aqara Smart Lock U200
+    vendorId: 0x115F
+    productId: 0x2802
+    deviceProfileName: lock-without-codes
```
Comment (lelandblue) [inline drivers/SmartThings/matter-lock/fingerprints.yml:7]: This profile is under discussion - Do not merge

**Comment by hcarter-775** [comment]
Can this ticket be closed, since we now have a fingerprint for this device?

**Comment by HunsupJung** [comment]
@hcarter-775 Yes we can close this ticket~!

**Pull request closed without merging**